src/shapes/Image-filters.spec.ts

```diff
@@ -0,0 +1,1906 @@
+        JSON.stringify(json),
+        'JSON representation should match expected value',
+      ).toBe('{"type":"Gamma","gamma":[1,1,1]}');
+
+      filter.gamma = [1.5, 1.5, 1.5];
+
+      json = filter.toJSON();
+
+      expect(
+        JSON.stringify(json),
+        'JSON representation with gamma [1.5, 1.5, 1.5] should match expected value',
+      ).toBe('{"type":"Gamma","gamma":[1.5,1.5,1.5]}');
+    });
+
+    it('fromObject', async () => {
+      const filter = new Gamma();
+      const object = filter.toObject();
+
+      const restoredFilter = await Gamma.fromObject(object);
+
+      expect(restoredFilter, 'restored filter should match original').toEqual(
+        filter,
+      );
+    });
+
+    it('isNeutralState', () => {
+      const filter = new Gamma();
+
+      expect(
+        filter.isNeutralState(),
+        'should be neutral when gamma is [1,1,1]',
+      ).toBeTruthy();
+
+      filter.gamma = [1.5, 1.5, 1.5];
+
+      expect(
+        filter.isNeutralState(),
+        'should not be neutral when gamma changes',
+      ).toBeFalsy();
+    });
+  });
+
+  describe('Convolute', () => {
+    it('constructor', () => {
+      expect(Convolute, 'Convolute filter should exist').toBeTruthy();
+
+      const filter = new Convolute();
+
+      expect(filter, 'should inherit from Convolute').toBeInstanceOf(Convolute);
+    });
+
+    it('properties', () => {
+      const filter = new Convolute();
+
+      expect(filter.type, 'type should be Convolute').toBe('Convolute');
+      expect(filter.opaque, 'default opaque should be false').toBe(false);
+      expect(
+        filter.matrix,
+        'default matrix should match expected values',
+      ).toEqual([0, 0, 0, 0, 1, 0, 0, 0, 0]);
+
+      const filter2 = new Convolute({
+        // @ts-expect-error -- TODO -- check this why 1 is passed instead of boolean
+        opaque: 0.5,
+        matrix: [1, -1, 1, 0, 1, 0, 0, 0, 0],
+      });
+
+      expect(filter2.opaque, 'opaque should match constructor value').toBe(0.5);
+      expect(filter2.matrix, 'matrix should match constructor value').toEqual([
+        1, -1, 1, 0, 1, 0, 0, 0, 0,
+      ]);
+    });
+
+    it('applyTo2d', () => {
+      const filter = new Convolute();
+
+      expect(filter.applyTo2d, 'should have applyTo2d method').toBeTypeOf(
+        'function',
+      );
+    });
+
+    it('toObject', () => {
+      // @ts-expect-error -- TODO -- check this why 1 is passed instead of boolean
+      const filter = new Convolute({ opaque: 1 });
+
+      expect(filter.toObject, 'should have toObject method').toBeTypeOf(
+        'function',
+      );
+
+      const object = filter.toObject();
+
+      expect(
+        JSON.stringify(object),
+        'serialized object should match expected value',
+      ).toBe('{"type":"Convolute","opaque":1,"matrix":[0,0,0,0,1,0,0,0,0]}');
+    });
+
+    it('toJSON', () => {
+      // @ts-expect-error -- TODO: check this -- it is typed as boolean but we are passing number here
```

Comment:
@asturur left a comment here in case you want to check it out
I assume boolean should be passed here instead of the number
or the type is wrong

there is also this same todo on 2 more places

Reply:
Yes the test was wrong, it makes sense as a boolean and the code behave as boolean.